runtime/frequency/src/lib.rs

```diff
@@ -1647,6 +1678,46 @@ impl pallet_utility::Config for Runtime {
 	type WeightInfo = weights::pallet_utility::SubstrateWeight<Runtime>;
 }
 
+parameter_types! {
+	pub MbmServiceWeight: Weight = Perbill::from_percent(80) * RuntimeBlockWeights::get().max_block;
+}
+
+impl pallet_migrations::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	type Migrations = (
+		pallet_stateful_storage::migration::v2::MigratePaginatedV1ToV2<
+			Runtime,
+			pallet_stateful_storage::weights::SubstrateWeight<Runtime>,
+		>,
+		pallet_stateful_storage::migration::v2::MigrateItemizedV1ToV2<
+			Runtime,
+			pallet_stateful_storage::weights::SubstrateWeight<Runtime>,
+		>,
+		pallet_stateful_storage::migration::v2::FinalizeV2Migration<
+			Runtime,
+			pallet_stateful_storage::weights::SubstrateWeight<Runtime>,
+		>,
+		pallet_messages::migration::MigrateV2ToV3<
+			Runtime,
+			pallet_messages::weights::SubstrateWeight<Runtime>,
+		>,
+		pallet_messages::migration::FinalizeV3Migration<
+			Runtime,
+			pallet_messages::weights::SubstrateWeight<Runtime>,
+		>,
+	);
+	// Benchmarks need mocked migrations to guarantee that they succeed.
+	#[cfg(feature = "runtime-benchmarks")]
+	type Migrations = pallet_migrations::mock_helpers::MockedMigrations;
+	type CursorMaxLen = ConstU32<65_536>;
+	type IdentifierMaxLen = ConstU32<256>;
+	type MigrationStatusHandler = ();
+	type FailedMigrationHandler = frame_support::migrations::FreezeChainOnFailedMigration;
```

Comment:
Q: Is there other strategies beside the freeze and what happens if in the worst case it get's frozen? Just a rough idea might be helpful to know

Reply:
consensus: `Ignore`